### PR TITLE
Fix #1588

### LINKED ISSUE #1588: Problems with tests selection from View All Tags dialog
There are some problems with the selection of tests from the View All Tags dialog.
(created a video [here](https://youtu.be/QEz6gQEGdbA))

- First access to View All Tags shows list in descending order.
- Selection of tests by tag is not always correct.

This is observed in RIDE 1.5.2.1 and 1.5.2 (when View All Tags sort feature was added #1565).

### PR DESCRIPTION
**Pull request by HelioGuilherme66**
Fixes tags selection from View All Tags dialog.
Thanks to @fzuellich for the main code refactoring.

### CODE REVIEW

File: src/robotide/ui/tagdialogs.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def __init__(self, controller, frame):
         self.unique_tags = 0
         self.total_test_cases = 0
         self.itemDataMap = dict()
-        self.sort_state = (1, 0)
+        self.sort_state = (0, 1)
```
Comment (yanne) [inline src/robotide/ui/tagdialogs.py]: What does this change mean?

File: src/robotide/ui/tagdialogs.py [hunk 2]
```diff
@@ -161,9 +157,11 @@ def _search_for_tags(self):
                     self._unique_tags[tag_name] = [test]
                     self._tagit[tag_name] = [tag]
 
+        isreversed = (self.sort_state[1] != 1)
         self.total_test_cases = len(self._test_cases)
-        self._results = sorted(
-            self._unique_tags.items(), key=lambda x: len(x[1]), reverse=True)
+        self._results = sorted(self._unique_tags.items(),
+                               key=lambda x: str(x[0]).lower, reverse=isreversed)
+        #                       key=lambda x: len(x[1]), reverse=isreversed)
```
Comment (yanne) [inline src/robotide/ui/tagdialogs.py]: commented code should be removed

File: src/robotide/ui/tagdialogs.py [hunk 3]
```diff
@@ -192,10 +190,12 @@ def OnClear(self, event):
         self._execute()
         for _, tests in self._results:
             self.tree.DeselectTests(tests)
+        """  Leave the tree as it was (Expanded/Collapsed)
```
Comment (yanne) [inline src/robotide/ui/tagdialogs.py]: Why is this code block commented? It should be removed if not needed.

File: src/robotide/ui/tagdialogs.py (hunk 1 above)
Comment (HelioGuilherme66) [inline src/robotide/ui/tagdialogs.py]: This change means, that initial sort order was descending on number of tests by tag, changed to tag names sorted ascending.

**Comment by fzuellich** [comment]
Ah oversaw that one. The change leads to sort by Name ascending. Before it was sorted by occurrence descending!

**Comment by yanne** [comment]
I cleaned up the commits, and merged this in a58a45311978e64

**Pull request closed without merging**